packages/ra-core/src/core/useConfigureAdminRouterFromChildren.spec.tsx

```diff
@@ -185,6 +189,30 @@ const TestedComponentWithPermissions = () => {
     );
 };
 
+const TestedComponentWithLazyCustomRoutes = ({ history }) => {
+    const [lazyRoutes, setLazyRoutes] = React.useState([]);
+
+    React.useEffect(() => {
+        setLazyRoutes([<Route path="/bar" element={<div>Bar</div>} />]);
+    }, [setLazyRoutes]);
+
+    return (
+        <CoreAdminContext history={history}>
+            <CoreAdminRoutes
+                layout={MyLayout}
+                catchAll={CatchAll}
+                loading={Loading}
+                ready={Ready}
+            >
+                <CustomRoutes noLayout>
+                    <Route path="/foo" element={<div>Foo</div>} />
+                </CustomRoutes>
+                <CustomRoutes>{...lazyRoutes}</CustomRoutes>
```

Comment:
Thanks for adding a test case.
However I don't fully understand it. Why don't you provide only dynamic routes?
Besides, when removing the fix, the test still passes, so I believe it should be written differently.

Can you fix it?
Thanks

Reply:
I have improved the test cases so when removing the fix the following case won't pass:
![Screenshot from 2022-12-14 14-19-04](https://user-images.githubusercontent.com/38666763/207705681-69d389e5-07dc-40a3-9e52-40d41d3f7dc9.png)